src/components/compress/index.tsx

```diff
@@ -232,6 +236,16 @@ export default class Compress extends Component<Props, State> {
     }
   }
 
+  componentDidMount() {
+    history.push(ROUTE_EDITOR);
+    history.on('popstate', this.props.onBack);
+  }
+
+  componentWillUnmount() {
+    history.replace(ROUTE_INDEX);
```

Comment:
I’m a bit confused by this one. `componentWillUnmount()` will happen when the user pressed back, so the route should already be correct, no?

Reply:
Correct, but the code in `componentWillUnmount()` covers a case, where the user presses a non-browser back button, the one render on top-left corner and triggers `onBack` function. If we didn't have the replace here then the history wouldn't update after clicking on it.

Reply:
Ah, you raise a good point. I think we shouldn’t be using `replaceState()` here, tho. Because it’s a back button, we should probably call `history.back()` when that button gets clicked.

So I’d vote for rejigging the button, remove the `replace()` here (and, by conclusion, remove `replace()` from the history module.

Reply:
Good idea. You can take a look now, I pushed an update where I trigger `history.back` after pressing the button  instead of using it in `componentWillUnmount()`.